### PR TITLE
feat(precompile): Add secp256r1 precompile

### PR DESCRIPTION
**Pull request by RomarQ**

### CODE REVIEW

File: frame/evm/precompile/secp256r1/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+extern crate alloc;
+
+use p256::ecdsa::{signature::hazmat::PrehashVerifier, Signature, VerifyingKey};
+use fp_evm::{ExitSucceed, PrecompileResult, Precompile, PrecompileHandle, PrecompileOutput};
+
+pub struct P256Verify;
+
+impl P256Verify {
+	/// https://github.com/ethereum/RIPs/blob/master/RIPS/rip-7212.md#precompiled-contract-gas-usage
+	const BASE_GAS: u64 = 3_450;
```
Comment (librelois) [inline frame/evm/precompile/secp256r1/src/lib.rs:31]: We should not use the base gas from the EIP as it was designed for a native execution.
We should add a benchmark scenario on the precompile-benchmark pallet to know how many weight to charge and charge with `record_external_cost`

File: frame/evm/precompile/secp256r1/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+		Ok(PrecompileOutput {
+			exit_status: ExitSucceed::Returned,
+			output: result.to_vec(),
+		})
+	}
+}
+
+/// (Signed payload) Hash of the original message
+/// 32 bytes of the signed data hash
+fn message_hash(input: &[u8]) -> &[u8] {
```
Comment (librelois) [inline frame/evm/precompile/secp256r1/src/lib.rs:64]: Please add the inline attribute for theses 3 functions, as there exist only to make the code more readable

File: frame/evm/precompile/secp256r1/src/lib.rs (hunk 1 above)
Comment (RomarQ) [reply frame/evm/precompile/secp256r1/src/lib.rs:31]: Fixed on moonbeam side

File: frame/evm/precompile/secp256r1/src/lib.rs (hunk 2 above)
Comment (RomarQ) [reply frame/evm/precompile/secp256r1/src/lib.rs:64]: Fixed on moonbeam side

**Comment by RomarQ** [comment]
Precompile moved to moonbeam side

**Pull request closed without merging**